app/src/main/java/com/kylecorry/trail_sense/tools/metaldetector/ui/FragmentToolMetalDetector.kt

```diff
@@ -61,6 +63,14 @@ class FragmentToolMetalDetector : BoundFragment<FragmentToolMetalDetectorBinding
 
     private val isMetalDetected = Debouncer(Duration.ofMillis(100))
 
+    private var whistle: ISoundPlayer = Whistle()
```

Comment:
Can you use this same technique to instantiate the Whistle?

On some devices it blocks the UI thread causing visible lag and on some devices it may even crash (only reported once, user reports their audio system is broken on their phone): https://github.com/kylecorry31/Trail-Sense/blob/083e708e005f087fb3e4ebddbb365b131bdc43b4/app/src/main/java/com/kylecorry/trail_sense/tools/whistle/quickactions/QuickActionWhistle.kt#L26